### ISSUE TITLE
Training 세트와 Test 세트를 분리하는 이유는?

### ISSUE THREAD

**Issue by jarammm** [AInsteins/AI-Tech-Interview#5]
2회차 - 2022년 8월 26일 금요일

**Comment by kimyuuuuuu** [comment]
- 모델은 데이터에 대해 예측값을 만들고 정답을 비교하며 학습.
- 학습데이터로 학습하더라도 실전에서 사용할때 성능을 잘 내야하기 때문에 학습과 테스트를 분리한다.
- 학습데이터만 사용하면 오버피팅 현상이 발생할 수 있다.

**Comment by miinkang** [comment]
- training은 모델을 학습하는 데 사용됩니다. 분리하지 않고 전체 데이터를 학습에 사용하게 되면, 모델의 성능을 평가할 수 없게 됩니다. 이미 학습한 데이터로 성능을 평가하는 것은 이미 답을 알고 있는 상태여서(seen data) cheating이 되기 때문입니다. 학습 후 성능을 평가하기 위해 training set, test set으로 분리합니다. 학습이 완료된 후 test set을 사용해 최종 모델 성능을 평가합니다. 

- test set 하나만을 갖고 모델을 평가하면서 모델의 성능을 개선한다면, test set에만 적합한 모델이 되어(over-fit) 범용적으로 사용이 어려울 수 있습니다. 이 때 validation set을 사용합니다. validation set은 검증을 위한 데이터 셋으로 학습이 된 여러 모델 중 가장 성능이 좋은 모델을 선택하기 위해 사용됩니다. validation set에 대한 정확도, 오차 등의 평가지표를 확인하면서 layer, hyper parameter등을 적절히 조절합니다.

**Comment by import-max** [comment]
학습 데이터에 대해서만 좋은 성능을 낸다면 좋은 모델이 아니다. overfitting이 된 모델이기 때문이다. 그렇기 때문에 처음 보는 데이터에 대해서도 좋은 성능을 내는지 판단하기 위해 데이터를 분리하여 사용한다.
